### DISCUSSION TITLE
randomB

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#156] [General]
Bert,

Continuing with the previous discussion. For constrained GLLVMs, I noticed that for 'nice' data sets, there are minimal differences between unregularised,  randomB = "LV", and  randomB = "P". 

For count data sets with lots of zeros, when fitting a constrained GLLVM without regularisation you sometimes get this 'quasi-complete separation' type pattern (very large negative estimated parameters + large SEs). It is not exactly quasi-complete separation...but it looks the same.

Regularisation then becomes useful.

And for highly collinear covariates with plenty of outliers....regularisation becomes rather important.

Then indeed...your last message....why not always do regularisation? Because comparing regularised and unregularised GLLVMs via an AIC...that is a dodgy comparison, isn't it? That is like comparing a GLM versus a GLLM via an AIC....which is a bad thing to do.

And if your covariates are all standardised....would you not always use  randomB = "LV"  ?  By default? That is cheaper in terms of sigmas.

The only advantage of randomB = "P" I can see is if you want to put a random effect inside the CLV. But I never understood why you want to do that.

So..the question is....are there any objections to using randomB = "LV" by default in constrained GLLVMs? At best it gives the same results as unregularised and at worst it avoids misery.

Alain

**Comment by BertvanderVeen** [comment]
Fixed effects are usually the starting place for most statistical models, so here too. That is not the place you need to start your analysis at, though.

`randomB="P"` and `randomB="LV"` can be seen as two different types of regularisation. "P" regularizes predictor-wise, so that all coefficients of predictors that are not important to the model are shrunk to near zero. "LV" shrinks LV-wise so that all predictor coefficients (constrained) per LV are shrunk to near zero. I'm not sure which of the two will stabilize the coefficients better, probably LV-wise.

`randomB="P"` could technically be used for a formula interface that allows a mix of fixed- and random-effects on the LVs. I might expand on this in the future, because if seen as a true random-effects formulation, combined with the concurrent ordination method the random-effects facilitate exploring the residual variation in the LVs, in contrast to the fixed-effects. They could then also be combined with correlation between the terms.

Finally, `randomB="P"` and` randomB="LV"` imply different correlation structures in terms of the species associations (see the new `getEnvironCov` and `getEnvironCor` in the development version of the package.

Ultimately, use of "randomB" is optional, because I want people to make a deliberate decision when using random-effects in a constrained/concurrent ordination (whether it is for regularisation purposes, or not), and the consequences that come with that. It means no confidence intervals (though see `getPredictErr`), a correlation structure, a (kind of) more complex model...
And generally, use of information criteria for random-effects models is a bit dodgy, and from a fixed vs. random effects angle the comparison might not be particularly helpful as you note.

The main argument against applying "randomB" by default, is that it might be overly conservative for small datasets, and for small effects, but I have not evaluated that in-depth with simulations.

**Reply by AlainZuur** [reply]
Hello Bert,

Thanks for the feedback. All clear.

Alain
